pyrdp/mitm/X224MITM.py

```diff
@@ -99,6 +99,12 @@ def onConnectionConfirm(self, _: X224ConnectionConfirmPDU):
         protocols = NegotiationProtocols.SSL if self.originalRequest.tlsSupported else NegotiationProtocols.NONE
 
         parser = NegotiationResponseParser()
+        response = parser.parse(pdu.payload)
+        if response.type == NegotiationType.TYPE_RDP_NEG_FAILURE:
```

Comment:
We took the habit to make these kind of checks with `isinstance(response, NegotiationFailurePDU)` whenever possible, since it allows IDEs with content-aware linting to know that it can assume that the `response` variable is indeed a `NegotiationFailurePDU` in this `if` block. 

With this current code, Pycharm would lint an error since it can't assume that `response` contains the `failureCode` attribute. I think it would also have spotted that the `NegotiationResponseParser.parse()` method has a return type of `NegotiationResponsePDU` and that the comparison is conflicting since the `NegotiationFailurePDU` does not inherit `NegotiationResponsePDU`

Reply:
Fixed